pilot/pkg/model/push_context.go

```diff
@@ -2421,6 +2421,30 @@ func (ps *PushContext) NetworkManager() *NetworkManager {
 	return ps.networkMgr
 }
 
+func (ps *PushContext) BestEffortInferHBONE(service *Service, port *Port) bool {
```

Comment:
~Might be worth a comment saying "This only returns true if all endpoints of the service support HBONE on the given port".~ Actually, I like the struct field name you use below; any reason you can't name the func `AllInstancesSupportHBONE`?